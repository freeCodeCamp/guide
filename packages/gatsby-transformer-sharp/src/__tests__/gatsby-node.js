const Promise = require(`bluebird`)

const { onCreateNode } = require(`../gatsby-node`)

describe(`Process image nodes correctly`, () => {
  it(`correctly creates an ImageSharp node from a file image node`, async () => {
    const node = {
      extension: `png`,
      id: `whatever`,
      children: [],
      internal: {
        contentDigest: `whatever`,
      },
    }
    const createNode = jest.fn()
    const createParentChildLink = jest.fn()
    const boundActionCreators = { createNode, createParentChildLink }

    await onCreateNode({
      node,
      boundActionCreators,
    }).then(() => {
      expect(createNode.mock.calls).toMatchSnapshot()
      expect(createParentChildLink.mock.calls).toMatchSnapshot()
      expect(createNode).toHaveBeenCalledTimes(1)
      expect(createParentChildLink).toHaveBeenCalledTimes(1)
    })
  })
})
