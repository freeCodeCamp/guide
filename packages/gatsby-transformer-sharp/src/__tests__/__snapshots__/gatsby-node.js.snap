// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Process image nodes correctly correctly creates an ImageSharp node from a file image node 1`] = `
Array [
  Array [
    Object {
      "children": Array [],
      "id": "whatever >> ImageSharp",
      "internal": Object {
        "contentDigest": "whatever",
        "mediaType": undefined,
        "type": "ImageSharp",
      },
      "parent": "whatever",
    },
  ],
]
`;

exports[`Process image nodes correctly correctly creates an ImageSharp node from a file image node 2`] = `
Array [
  Array [
    Object {
      "child": Object {
        "children": Array [],
        "id": "whatever >> ImageSharp",
        "internal": Object {
          "contentDigest": "whatever",
          "mediaType": undefined,
          "type": "ImageSharp",
        },
        "parent": "whatever",
      },
      "parent": Object {
        "children": Array [],
        "extension": "png",
        "id": "whatever",
        "internal": Object {
          "contentDigest": "whatever",
        },
      },
    },
  ],
]
`;
